File: Python/Easy/Minimum Waiting Time/solution_1.py

lst = [3, 2, 1, 2, 6]


def minimumWaitingTime(queries):

    hash_tbl = {}
    queries.sort()
    sums = 0

    for i in range(len(queries) - 1):
        first = queries[i]
        second = queries[i + 1]
        sums += first

        hash_tbl[sums] = second
        
    
    return sum(hash_tbl.keys())

print(minimumWaitingTime(lst))

File: Go/Easy/Minumum Waiting Time/solution_1.go
package main

import "sort"

func MinimumWaitingTime(queries []int) int {

	hash_tbl := make(map[int]int)
	sort.Ints(queries)
	waitingTime := 0
	totalWaitingTime := 0

	for idx := 0; idx < len(queries)-1; idx++ {

		first := queries[idx]
		second := queries[idx+1]

		waitingTime += first

		hash_tbl[waitingTime] = second

	}

	for key := range hash_tbl {
		totalWaitingTime += key

	}

	return totalWaitingTime

}
